src/machine/machine_nrf52xxx.go

```diff
@@ -183,7 +183,12 @@ func (a *ADC) Get() uint16 {
 		resolutionAdjustment = 4 // 12bit
 	}
 
-	return rawValue.Get() << resolutionAdjustment
+	value := int16(rawValue.Get())
```

Comment:
This should probably be a cast to `uint16`

Reply:
This is how it was before, see https://github.com/tinygo-org/tinygo/pull/4701/files#diff-e8032f214fa1ae692ba79fc6179cd57e7325ec7d8da3152ff0e4223587f1a041L167

Reply:
Isn't that previous cast to int16?

```go
value := int16(rawValue.Get())
```